### PR TITLE
Visualize source maps in tests

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | `Fixes #1, Fixes #2` 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

Source maps are difficult to debug, especially because our current snapshots are useless. I built a small tool that is like source-map-visualizer, but based on text rather than interactive.

I only enabled it for one test, but I could also enable it automatically for all the tests that use source maps.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/15931"></a>

### CODE REVIEW

File: packages/babel-core/test/fixtures/transformation/source-maps/arrow-function/source-map-visual.txt [hunk 1]
```diff
@@ -0,0 +1,41 @@
+    (1:0) var t         <--     (1:0) var t
```
Comment (nicolo-ribaudo) [inline packages/babel-core/test/fixtures/transformation/source-maps/arrow-function/source-map-visual.txt:1]: The format is `(line:column) SOURCE <-- (line:column) OUTPUT`

**Comment by nicolo-ribaudo** [comment]
What format would you prefer?

(a) mappings point at a character
```
    (1:4) var t = x     <--     (1:4) var t = f
              ^                           ^

    (1:5) ar t = x      <--     (1:5) ar t = fu
              ^                           ^
```

(b) mappings point between two character (note: this inserts a `|` in the code)
```
    (1:4) var |t = x     <--     (1:4) var |t = f
              |                            |

    (1:5) ar t| = x      <--     (1:5) ar t| = fu
              |                            |
```

File: packages/babel-core/test/fixtures/transformation/source-maps/arrow-function/source-map-visual.txt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+              ^                           ^
+
+    (1:8) t = x =>      <--    (1:19) n t(x) {
+              ^                           ^
+
+    (1:9)  = x => x     <--    (1:20)  t(x) {
+              ^                           ^
+
+    (1:9)  = x => x     <--     (2:2)   retur
+              ^                         ^
```
Comment (nicolo-ribaudo) [inline packages/babel-core/test/fixtures/transformation/source-maps/arrow-function/source-map-visual.txt:26]: This PR is just for adding tests and fixes will be in follow-ups, but this is weird. The `r` should either be mapped to `=>`'s `=`, or to the `x` after `=>`.

**Comment by nicolo-ribaudo** [comment]
Actually, I prefer this with ranges rather than just markers:
```
  (1:0-4) var t =       <--   (1:0-4) var t = 
          ^^^^                        ^^^^

  (1:4-5) var t = x     <--   (1:4-5) var t = f
              ^                           ^

  (1:5-8) ar t = x =>   <--   (1:5-8) ar t = func
              ^^^                         ^^^

    (1:8) t = x =>      <--  (1:8-17) t = function (x) 
             ><                           ^^^^^^^^^

    (1:8) t = x =>      <-- (1:17-18) ion (x) {
             ><                           ^

  (1:8-9) t = x =>      <-- (1:18-19) on (x) {
              ^                           ^

 (1:9-13)  = x => x *   <--   (2:2-9)   return x * 
              ^^^^                      ^^^^^^^

(1:13-14)  => x * x     <--  (2:9-10) urn x * x
              ^                           ^

(1:14-17) => x * x;     <-- (2:10-13) rn x * x;
              ^^^                         ^^^

(1:17-18) x * x;        <-- (2:13-14) x * x;
```

File: packages/babel-helper-fixtures/src/index.ts [hunk 3]
```diff
@@ -51,6 +53,7 @@ export interface TaskOptions extends InputOptions {
   ignoreOutput?: boolean;
   minNodeVersion?: string;
   sourceMap?: boolean;
+  sourceMapVisual?: boolean;
```
Comment (nicolo-ribaudo) [inline packages/babel-helper-fixtures/src/index.ts:56]:
```suggestion
```

File: packages/babel-core/test/fixtures/transformation/source-maps/input-source-map-complex/source-map-visual.txt [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+          ^^^^^^^^^^^                                ^^^
+
+(90:0-12)     return {            <--  (95:4-12)     return {
+          ^^^^^^^^^^^^                               ^^^^^^^^
+
+(91:0-39)       msg: 'Welcome to Your Vue.js App' <--   (96:6-9)     msg: 'W
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                    ^^^
+
+(91:0-39)       msg: 'Welcome to Your Vue.js App' <--  (96:9-39)  msg: 'Welcome to Your Vue.js App'
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
[the remark this answers is not in the document]
Comment (nicolo-ribaudo) [reply packages/babel-core/test/fixtures/transformation/source-maps/input-source-map-complex/source-map-visual.txt:23]: Yep, this file is all very weird. It's the result of merging the generated source map with an existing one, so the problem might be there.

**Review by liuxingbaoyu — APPROVED** [review]
Amazing!
Perhaps we can add visualization of `identifier names` in the future.

**Pull request merged**